File: easy/word_to_digit.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

func main() {
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()

	digits := map[string]int{"zero": 0, "one": 1, "two": 2,
		"three": 3, "four": 4, "five": 5, "six": 6,
		"seven": 7, "eight": 8, "nine": 9}

	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		t := strings.Split(scanner.Text(), ";")
		for _, i := range t {
			fmt.Print(digits[i])
		}
		fmt.Println()
	}
}

File: easy/word_to_digit.hs
import System.Environment (getArgs)
import Data.List.Split (splitOn)
import Data.Maybe (fromJust)

digits :: [(String, Int)]
digits = [("one", 1), ("two", 2), ("three", 3), ("four", 4), ("five", 5), ("six", 6), ("seven", 7), ("eight", 8), ("nine", 9), ("zero", 0)]

doLookup    :: [(String, Int)] -> String -> String
doLookup d l = show . fromJust $ lookup l d

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (concatMap (doLookup digits) . splitOn ";") $ lines input
